service/src/main/kotlin/fi/espoo/evaka/invoicing/controller/FinanceBasicsController.kt

```diff
@@ -154,6 +157,35 @@ class FinanceBasicsController(
             }
             .also { Audit.FinanceBasicsVoucherValuesRead.log() }
     }
+
+    @DeleteMapping("/voucher-values/{id}")
+    fun deleteVoucherValue(
+        db: Database,
+        user: AuthenticatedUser.Employee,
+        clock: EvakaClock,
+        @PathVariable id: ServiceNeedOptionVoucherValueId,
+    ) {
+        return db.connect { dbc ->
+                dbc.transaction { tx ->
+                    accessControl.requirePermissionFor(
+                        tx,
+                        user,
+                        clock,
+                        Action.Global.DELETE_VOUCHER_VALUE
+                    )
+
+                    val values = tx.getServiceNeedVoucherValuesByVoucherValueRangeId(id)
+
+                    if (values.isEmpty()) throw NotFound("Voucher value $id not found")
+                    if (values[0].id != id)
+                        throw BadRequest("Can only delete the latest voucher value")
+
+                    tx.deleteVoucherValue(id)
+                    tx.reopenVoucherValueValidityRange(values[1].id)
```

Comment:
doesn't this go boom if you delete the only one (size=1)?

Reply:
Indeed, good catch. Added a test for this and fixed.